x-pack/plugins/endpoint/server/services/endpoint/alert_query_builders.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { KibanaRequest } from 'kibana/server';
+import { EndpointAppConstants } from '../../../common/types';
+import { EndpointAppContext } from '../../types';
+
+export const kibanaRequestToAlertListQuery = async (
+  pagingProperties: Record<string, any>,
+  endpointAppContext: EndpointAppContext
+): Promise<Record<string, any>> => {
+  const DEFAULT_TOTAL_HITS = 10000;
+
+  // Calculate minimum total hits set to indicate there's a next page
+  const totalHitsMin = Math.max(
+    pagingProperties.fromIndex + pagingProperties.pageSize * 2,
+    DEFAULT_TOTAL_HITS
+  );
+
+  return {
+    body: {
+      track_total_hits: totalHitsMin,
+      query: {
+        match_all: {},
+      },
+      sort: [
+        {
+          '@timestamp': {
+            order: 'desc',
+          },
+        },
+      ],
+    },
+    from: pagingProperties.fromIndex,
+    size: pagingProperties.pageSize,
+    index: EndpointAppConstants.ALERT_INDEX_NAME,
+  };
+};
+
+export const getPagingProperties = async (
+  request: KibanaRequest<any, any, any>,
+  endpointAppContext: EndpointAppContext
+): Promise<Record<string, any>> => {
```

Comment:
Does this need to return a `Promise`?

Reply:
IMO we should almost always use async functions to allow for context switching, no?

Reply:
Doesn't it have to be an `async` function because it's doing an `await` below?

Reply:
I don't see any uses of `await` inside this function. But yes, if it had an `await` in there, we would have to define the function as `async`.

Reply:
there is an await for the config which is an observable.